lib/charms/filesystem_client/v0/filesystem_info.py

```diff
@@ -291,19 +293,19 @@ def _hostinfo(host: str) -> tuple[str, Optional[int]]:
     # more characters after the hostname <==> port
 
     if host[pos] != ":":
-        raise ParseError("expected `:` after IPv6 address")
+        raise ParseUriError("expected `:` after IPv6 address")
     try:
         port = int(host[pos + 1 :])
     except ValueError:
-        raise ParseError("expected int after `:` in host")
+        raise ParseUriError("expected int after `:` in host")
 
     return hostname, port
 
 
-T = TypeVar("T", bound="FsInfo")
+T = TypeVar("T", bound="FilesystemInfo")
```

Comment:
Hmm, I'm a bit neutral about this. A lot of charms will probably differ on their support of noble (e.g. cephfs-server-proxy and nfs-server-proxy are still on Jammy), so do we want to force the use of Noble on all the charms that use this library? IIRC Azure only supports Lustre on Noble, so it would be sad to have to downgrade this in the future if we determine that it's needed there.

Reply:
Let's keep `T = TypeVar(...)` for now then.

We could use the `typing-extensions` package to bring in the `Self` type if we choose to continue supporting Jammy. We're already going to be doing an awkward dance with Lustre since the Slurm charms are now on Noble so we can bring in other packages we need like `sackd` and other special sauce. However, I'd rather not add an external dependency to the charm library just to use a special type alias.